File: Exercism Codes/JavaScript/diamond.js
export const rows = (letter) => {
  const result = [];

  for (let x = 65; x <= letter.charCodeAt(0); x++) {
    result.push(line(letter, x));
  }

  for (let x = letter.charCodeAt(0) - 1; x >= 65; x--) {
    result.push(line(letter, x));
  }

  return result;
};

const line = (letter, x) => {
  const pad1 = " ".repeat((letter.charCodeAt(0) - 65) - (x - 65));
  const pad2 = " ".repeat(x - 65);
  const firstHalfLine = pad1 + String.fromCharCode(x) + pad2;
  const secondHalfLine = pad2 + String.fromCharCode(x) + pad1;
  return firstHalfLine + secondHalfLine.substring(1);
}

File: Exercism Codes/Python/diamond.py
def rows(letter):
    result = []
    for x in range(ord("A"), ord(letter) + 1):
        result.append(line(letter, x))

    for x in range(ord(letter) - 1, ord("A") - 1, -1):
        result.append(line(letter, x))

    return result

def line(letter, x):
    pad_1 = (" " * ((ord(letter) - ord("A")) - (x - ord("A"))))
    pad_2 = (" " * (x - ord("A")))
    first_half_line = pad_1 + chr(x) + pad_2
    second_half_line = pad_2 + chr(x) + pad_1
    return first_half_line + second_half_line[1:]